x/pocketcore/types/session_test.go

```diff
@@ -35,216 +43,462 @@ func TestSessionKey_Validate(t *testing.T) {
+		UnstakingCompletionTime: time.Time{},
+	}
+	node6 := nodesTypes.Validator{
+		Address:                 sdk.Address(fakePubKey6.Address()),
+		PublicKey:               (fakePubKey6),
+		Jailed:                  false,
+		Status:                  sdk.Staked,
+		Chains:                  []string{ethereum},
+		ServiceURL:              "https://www.google.com:443",
+		StakedTokens:            sdk.NewInt(100000),
+		UnstakingCompletionTime: time.Time{},
+	}
+	node7 := nodesTypes.Validator{
+		Address:                 sdk.Address(fakePubKey7.Address()),
+		PublicKey:               (fakePubKey7),
+		Jailed:                  false,
+		Status:                  sdk.Staked,
+		Chains:                  []string{ethereum},
+		ServiceURL:              "https://www.google.com:443",
+		StakedTokens:            sdk.NewInt(100000),
+		UnstakingCompletionTime: time.Time{},
+	}
+	node8 := nodesTypes.Validator{
+		Address:                 sdk.Address(fakePubKey8.Address()),
+		PublicKey:               (fakePubKey8),
+		Jailed:                  false,
+		Status:                  sdk.Staked,
+		Chains:                  []string{ethereum},
+		ServiceURL:              "https://www.google.com:443",
+		StakedTokens:            sdk.NewInt(100000),
+		UnstakingCompletionTime: time.Time{},
+	}
+	node9 := nodesTypes.Validator{
+		Address:                 sdk.Address(fakePubKey9.Address()),
+		PublicKey:               (fakePubKey9),
+		Jailed:                  false,
+		Status:                  sdk.Staked,
+		Chains:                  []string{ethereum},
+		ServiceURL:              "https://www.google.com:443",
+		StakedTokens:            sdk.NewInt(100000),
+		UnstakingCompletionTime: time.Time{},
+	}
+	node10 := nodesTypes.Validator{
+		Address:                 sdk.Address(fakePubKey10.Address()),
+		PublicKey:               (fakePubKey10),
+		Jailed:                  false,
+		Status:                  sdk.Staked,
+		Chains:                  []string{ethereum},
+		ServiceURL:              "https://www.google.com:443",
+		StakedTokens:            sdk.NewInt(100000),
+		UnstakingCompletionTime: time.Time{},
+	}
+	node11 := nodesTypes.Validator{
+		Address:                 sdk.Address(fakePubKey11.Address()),
+		PublicKey:               (fakePubKey11),
+		Jailed:                  false,
+		Status:                  sdk.Staked,
+		Chains:                  []string{ethereum},
+		ServiceURL:              "https://www.google.com:443",
+		StakedTokens:            sdk.NewInt(100000),
+		UnstakingCompletionTime: time.Time{},
+	}
+	allNodes = make([]exported.ValidatorI, 12)
+	allNodes[0] = node12
+	allNodes[1] = node1
+	allNodes[2] = node2
+	allNodes[3] = node3
+	allNodes[4] = node4
+	allNodes[5] = node5
+	allNodes[6] = node6
+	allNodes[7] = node7
+	allNodes[8] = node8
+	allNodes[9] = node9
+	allNodes[10] = node10
+	allNodes[11] = node11
+	sessionCtx := newContext(t, false)
+	ctx := newContext(t, false)
+	k := MockPosKeeper{Validators: allNodes}
+	sessionNodes, err := NewSessionNodes(
```

Comment:
If you just add a comment saying "context before upgrade" and "context after upgrade". Are we going to need the copy-paste of all the preparation?